cmd/kas-fleet-manager/main_test.go

```diff
@@ -19,7 +20,9 @@ func TestInjections(t *testing.T) {
 	)
 	g.Expect(err).To(gomega.BeNil())
 	err = env.CreateServices()
-	g.Expect(err).To(gomega.BeNil())
+	if err != nil && !strings.Contains(err.Error(), "error validating GCP API credentials") {
```

Comment:
Maybe we could play with `ProvideValue` instead to provide a hardcoded value for the providers in this test. Could you explore that potential approach?

Reply:
I've experimented with the dependency injection and `ProvideValue` a bit, but I don't think it is useful in this case. `ProvideValue` can be used to provide an additional instance of the GCP config with pre-populated values that pass validation.

However, this only provides an additional instance, validation will still fail on the original GCP config that was provided in `ConfigProviders`. Using tags to exclude the GCP config only from this test doesn't seem to be practical because `goava` doesn't allow you to have 'not' conditions on tags. So you can't express something like `always provide this, except if a tag has a certain value`. At least I didn't find a way to do that.

So conclusion: checking the error string seems to be the most practical option.

Reply:
:+1: I see. it's unfortunate that the DI framework does not provide that.
The test is quite fragile with this change (if the error string value changes the test will break, etc...) but for now it will work to workaround the DI framework / mechanism we have on top of it.